maint/bumplibs.sh

```diff
@@ -177,10 +181,14 @@ process_lib() {
     fi
     git --no-pager diff --color -w $LAST_RELEASE..HEAD $HEADERS_HEAD
 
-    # Show commits touching lib since last release
     echo ""
-    echo "- Commits (without Refactor & Build) touching lib$LIB since $LAST_RELEASE:"
-    git log --color Pacemaker-2.0.3..HEAD -z $HEADERS_HEAD $SOURCES $AMFILE|grep -vzE "Refactor:|Build:|Merge pull request"
+    if yesno "Show commits (minus refactor/build/merge) touching lib$LIB since $LAST_RELEASE [y/N]?"
```

Comment:
Oops another leftover ...
Personally that many yes/no continue prompts do confuse me and make me press something wrong where I don't have an easy way to go back one step here.
Maybe we can address this somehow in the next iteration.
Don't know if it is worth going for curses to be able to fold/unfold sections with a key.
Wouldn't necessarily have to stay a shell script either.
Non-interactive report mode to get a nice document would be cool as well.

Reply:
Definitely, it's awkward.

Maybe for each library, it could generate a (temporary) text file report, and open it with $EDITOR, with the first line being the compatible/incompatible/fixes choice. The user would browse the report, edit the first line with their choice, and exit the editor. The script would grab the choice and propose the new version, and the user could accept, quit, or redo.

Might be nice to generate the report with markdown and set $EDITOR to a markdown editor :)

Reply:
Hmm ... did we talk about this.
Now that you come up with that idea I remember that I had thought about playing with something like that when I would be doing the next release - latest ;-)
Could imagine to have everything in just a single file that you create using bumplibs.sh.
Afterwards edit with your editor of choice.
And have it processed with another call of bumplibs.sh.
That would as well remove the need for checking if we have an interrupted and restarted run or something.